src/fenicsxconcrete/boundary_conditions/bcs.py

```diff
@@ -36,8 +38,8 @@ def __init__(
         self,
         domain: dolfinx.mesh.Mesh,
         space: dolfinx.fem.FunctionSpace,
-        facet_tags: typing.Optional[npt.NDArray] = None,
-    ):
+        facet_tags: np.ndarray | None = None,
```

Comment:
isn't `np.ndarray | None` redundant if the default value is `None`?
also what is usually recommended, `np.ndarray` or the more generic `numpy.typing.NDArray`?

Reply:
I would also say that the None is redundant. I think the optional type which was used before is useful in strongly typed languages but I don't see the benefit of having this `np.ndarray | None` here

Reply:
I would suggest use NDArray only if we explicitly give a dtype in brackets: like  npt.NDArray[np.float64]

Reply:
remark: from PEP604 (https://peps.python.org/pep-0604/):
"Optional values should be equivalent to the new union syntax
None | t == typing.Optional[t]"

Before you had Optional(array) which now would be array | None

Reply:
> remark: from PEP604 (https://peps.python.org/pep-0604/):
"Optional values should be equivalent to the new union syntax
None | t == typing.Optional[t]"

this means we can drop the `= None` and simply write `np.ndarray | None`, right?

Reply:
No, `ndarray | None` is just the type hint and it does not give us the default value,
````python
def optional(x : np.ndarray | None) -> None:
    print(x)

optional()
```

would raise an exception, because the default value was not specified

Reply:
yes to @srosenbu 
we need both: all types and the default value if it is like a kwarg argument